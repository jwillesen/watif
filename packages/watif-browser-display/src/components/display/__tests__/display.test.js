import React from 'react'
import Display from '../index'
import {shallow} from 'enzyme'

it('renders', () => {
  const wrapper = shallow(<Display storyState={{}} />)
  expect(wrapper).toMatchSnapshot()
})

it('calls calls executeVerb with examine verb on item click', () => {
  const executeVerb = jest.fn()
  const wrapper = shallow(<Display
    storyState={{}}
    executeVerb={executeVerb}
  />)
  wrapper.instance().handleItemClick('some-item-id')
  expect(executeVerb).toHaveBeenCalledWith({
    id: 'examine', subject: 'some-item-id', target: null,
  })
})
