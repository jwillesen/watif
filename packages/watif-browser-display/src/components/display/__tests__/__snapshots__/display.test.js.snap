// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  styleName="root"
>
  <div
    styleName="inventory-tree"
  >
    <InventoryTree />
  </div>
  <div
    styleName="story-content"
  >
    <StoryContent
      onItemClick={[Function]}
      storyState={Object {}}
    />
  </div>
</div>
`;
